### ISSUE TITLE
Weird behaviour using lists in variables & outputs

### ISSUE THREAD

**Issue by antonosmond** [hashicorp/terraform#8550]
### Terraform Version
0.7.2 (on OS X)

### Affected Resource(s)
- outputs

### Terraform Configuration Files
```hcl
variable "somemap" {
  default = {
    somestring = "hello world"
    somelist = [
      "one",
      "two",
      "three"
    ]
  }
}

output "something" {
  value = "${var.somemap["somestring"]}"
}
```

### Expected Behavior
The value of output 'something' shoud equal 'hello world'  

### Actual Behavior
No output created and no crash or error message

### Steps to Reproduce
1. `terraform apply`

### Important Factoids
If I remove the list from the variable, it works as expected:
```hcl
variable "somemap" {
  default = {
    somestring = "hello world"
  }
}

output "something" {
  value = "${var.somemap["somestring"]}"
}
```
```
Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

something = hello world
```
If I reference the entire map in the output  the map is output and looks sensible:
```hcl
variable "somemap" {
  default = {
    somestring = "hello world"
    somelist = [
      "one",
      "two",
      "three"
    ]
  }
}

output "something" {
  value = "${var.somemap}"
}
```
```
Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

something = {
  somelist = [one two three]
  somestring = hello world
}
```

Basically, if my map variable contains a list, it breaks any outputs based on that map. I imagine I can work around it for now by using a comma separated string instead of a list but then I'll be back to lots of joins & splits like in the older terraform versions.

**Comment by jbardin** [comment]
Hi @antonosmond,

Thanks for filing this issue. The problem here stems from maps currently requiring homogenous types. Having the mix of a string and list types is causing an error, but that error in an "output" fails silently. If you tried to use a map like this in a resource, you would see

map "var.somemap" does not have homogenous value types

I opened an separate issue to track the lack of feedback from outputs here at #9568
